controller/api/destination/server.go

```diff
@@ -259,25 +259,18 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 		// name. When we fetch the profile using a pod's DNS name, we want to
 		// return an endpoint in the profile response.
 		if hostname != "" {
-			pod, err := getPodByHostname(s.k8sAPI, hostname, service)
+			address, pod, err := s.getEndpointByHostname(s.k8sAPI, hostname, service, port)
```

Comment:
Instead of `getEndpointByHostname` returning an address and a Pod, we could return just the address. When we create the address, if it's backed by a Pod, we can access that through `address.Pod` in the rest of this function. When it's `nil`, it will be treated just the same as when we return a non-`nil` address and a `nil` Pod.

Reply:
good call